### PR TITLE
CMake Super-Build for Bio-Formats 5.1.4

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by hflynn** [comment]
You need to update the icons for the API docs and source code links too. Images already added in #158 you just need to update the image names.

**Comment by hflynn** [comment]
Ah actually, there are some extra source code links here that I'm not sure Gus made icons for

File: bf_cpp_downloads.html [hunk 1]
```diff
@@ -280,12 +350,21 @@
...
-                                        alt="Zipped Source Code" /></a></td>
+                                        alt="Zipped Bio-Formats Source Code" /></a></td>
                                 <td align="center">@SOURCE_CODE_SIZE@</td>
                                 <td>@SOURCE_CODE_BASE@</td>
                                 <td align="center">@SOURCE_CODE_SHA1@ (<a
                                         href="@SOURCE_CODE@.sha1">SHA1</a>)</td>
                             </tr>
+                            <tr>
+                                <td><a href="@SUPERBUILD_SOURCE_CODE@" target="_blank"><img
+                                        src="images/bf_source_code.png"
```
Comment (hflynn) [inline bf_cpp_downloads.html]: Gus made a new icon for this which you haven't added (bf_super-build_source_code.png)

File: bf_cpp_downloads.html [hunk 2]
```diff
@@ -202,38 +164,146 @@
...
+                           platform release and therefore should
+                           always work with the shared libraries
+                           provided by the platform.  These releases
+                           are provided only for platforms with
+                           guaranteed ABI stability during their
+                           supported lifetime, and which also provide
+                           all of the needed build dependencies.</p>
                     </li>
                 </ul>
                 <h2><a name="api" id="api"></a>Bio-Formats C++ API
```
Comment (hflynn) [inline bf_cpp_downloads.html:305]: Image name for API docs icon below needs changing to use new icon:
bf_api_docs.png -> bf_api_doc.png

File: bf_cpp_downloads.html [hunk 3]
```diff
@@ -311,7 +397,12 @@
                             <tr>
                                 <td><a href="@TAG_URL@" target="_blank"><img
                                         src="images/bf_gh_tag.png"
```
Comment (hflynn) [inline bf_cpp_downloads.html]: image name needs changing:
bf_gh_tag.png -> git_tag.png

File: bf_cpp_downloads.html [hunk 4]
```diff
@@ -311,7 +397,12 @@
...
-                                        alt="GitHub Tag" /></a></td>
+                                        alt="Bio-Formats GitHub Tag" /></a></td>
+                            </tr>
+                            <tr>
+                                <td><a href="@SUPERBUILD_TAG_URL@" target="_blank"><img
+                                        src="images/bf_gh_tag.png"
```
Comment (hflynn) [inline bf_cpp_downloads.html]: As above, needs changing to git_tag.png

File: bf_cpp_downloads.html [hunk 5]
```diff
@@ -300,7 +379,14 @@
...
+                            </tr>
+                            <tr>
+                                <td><a
+                                        href="http://github.com/ome/ome-cmake-superbuild"
+                                        target="_blank"><img
+                                        src="images/bf_gh_repo.png"
+                                        alt="CMake Super-Build GitHub Repository Link" /></a></td>
                             </tr>
                             <tr>
                                 <td><a href="http://git.openmicroscopy.org/"
```
Comment (hflynn) [inline bf_cpp_downloads.html:388]: images/bf_gh_web.png (line 394 below) needs changing to images/gitweb_interface.png

File: bf_cpp_downloads.html [hunk 6]
```diff
@@ -300,7 +379,14 @@
                                         href="http://github.com/openmicroscopy/bioformats"
                                         target="_blank"><img
                                         src="images/bf_gh_repo.png"
-                                        alt="GitHub Repository Link" /></a></td>
```
Comment (hflynn) [inline bf_cpp_downloads.html:303]: images/bf_gh_repo.png needs replacing with images/github_repository.png

File: bf_cpp_downloads.html [hunk 7]
```diff
@@ -300,7 +379,14 @@
...
+                                        alt="Bio-Formats GitHub Repository Link" /></a></td>
+                            </tr>
+                            <tr>
+                                <td><a
+                                        href="http://github.com/ome/ome-cmake-superbuild"
+                                        target="_blank"><img
+                                        src="images/bf_gh_repo.png"
```
Comment (hflynn) [inline bf_cpp_downloads.html]: images/bf_gh_repo.png needs replacing with images/github_repository.png

**Comment by hflynn** [comment]
In line comments above will fix existing broken images then you have one more to add. We might also want to ask @gusferguson to make a couple of new icons if you want different ones for the github links.

**Comment by gusferguson** [comment]
@rleigh-dundee @hflynn - what icons do you need?

**Comment by ghost** [comment]
@hflynn All your suggested changes should now be live.

@gusferguson If you look at the bottom of the page, the "GitHub Repository" and "Git Tag" links are ambiguous due to having two of each.  Ideally we would have
- Bio-Formats Repository
- Super-Build Repository
- Bio-Formats Git Tag
- Super-Build Git Tag

and also
- Bio-Formats Source Code

for clarity and consistency above these (and also to match the documentation).

**Comment by hflynn** [comment]
Yep, all fixed. 

I agree it would be good to have the github and git tag icons updated to differentiate between BF and the superbuild. I'm not sure adding Bio-Formats to the source code link is really needed as it already has the BF logo on it and is exactly the same download as on the main BF download page with this same icon. However, I'll defer to @gusferguson on that point if he wants to update it with more text.

**Comment by ghost** [comment]
@hflynn Well, with respect to the icons, it's placing a lot of faith in the end user knowing what all those tiny icons actually mean, and in this case they are also identical for both cases, so I think having it explicitly spelled out would be better, particularly when we link here from the documentation telling the user to download one of these depending upon their needs.

**Comment by gusferguson** [comment]
@rleigh-dundee @hflynn - I will do all the buttons as suggested then you can see what combination works. Have them done later today - Thursday.

**Comment by gusferguson** [comment]
New buttons:

bf_bioformats_repository
bf_bioformats_source_code
bf_git_tag
bf_super-build_git_tag
bf_super-build_repository

**Comment by ghost** [comment]
Thanks @gusferguson.  I have updated the page with these buttons, so should be complete unless any further corrections or rewording is required.  @hflynn ?

**Comment by hflynn** [comment]
Good to merge from me - all icons correct, reads fine and all the links go to the right thing.

**Comment by ghost** [comment]
@gusferguson For the 5.1.5 release, could we possibly have some "MacOS 10.11" buttons to match the existing ones for 10.9 and 10.10?

**Comment by joshmoore** [comment]
Various small comments:
- http://downloads.openmicroscopy.org/bio-formats-cpp/5.1.4/artifacts/to-be-deleted/ needs cleaning up
- I can't remember exactly how, but I think it's possible to have a blurb on pages like http://downloads.openmicroscopy.org/bio-formats-cpp/5.1.4/artifacts/platform/, that might be useful to distinguish what's going on since we didn't find names for those directories that we were very happy with
- comparing with the java downloads page this is consistent, but for me, the explanations of what's what come low on the page (i.e. under the links). Not sure if that's something to address now or more generally later.

**Comment by ghost** [comment]
WRT the cleanup, we need to wipe the directory out completely and run the release jobs through from scratch to do a proper verification of everything from the release trigger.

For the explanations, this is the same as what was done for previous releases, and I think we discussed this before for 5.1.0?  We can move the description from after the file lists to before; the tradeoff is that if the reader just wants a file, it gets in the way of that since they would then have to navigate past it, which might get annoying.  It's easy enough to rearrange the order if needed, but if you're not sure what you want it's only a matter of looking further down.  What's more important to have first; a list of available platforms or a detailed explanation?  If you have to read the detail before looking further and realising none are applicable it's not too helpful.

**Comment by joshmoore** [comment]
> If you have to read the detail before looking further and realising none are applicable it's not too helpful.

Not necessarily saying everything should be above. But the page as it stands does assume you know what the titles mean.

**Comment by gusferguson** [comment]
@rleigh-dundee - no problem - will do ASAP.

**Comment by gusferguson** [comment]
@rleigh-dundee @hflynn 

bf_c _mac_10_11_debug
bf_c _mac_10_11_release

**Comment by ghost** [comment]
@gusferguson Thanks.  One additional point I missed: would it be possible to have variants of the source code buttons: the "Source Code", "Bio-Formats Source Code" and "Super-Build Source Code" buttons all have "zip" icons; would it be possible to have all three with "tar" in place of "zip"?

The "Super-Build" one only needs the "tar" variant; the others need both.  Thanks!

**Comment by gusferguson** [comment]
@rleigh-dundee - .tar, .txz, or .tgz?

**Comment by hflynn** [comment]
They are tar.xz files in the filename @gusferguson so tzx. sorry I missed that when I reviewed it.

**Comment by gusferguson** [comment]
@hflynn - cheers. @rleigh-dundee 

bf_bioformats_source_code_txz
bf_source_code_txz
bf_super-build_source_code_txz

**Comment by hflynn** [comment]
Sorry @gusferguson the 3rd one with the wrong extension on it is the API docs icon

**Comment by ghost** [comment]
@gusferguson Thanks.

**Comment by gusferguson** [comment]
@hflynn 

bf_api_doc_txz

**Comment by ghost** [comment]
@gusferguson Sorry to be a pain, but the buttons you did recently have a white rather than transparent background which looks odd on the page where there's a colour behind.  Would it be possible to re-save them with a transparent background?  This includes all of the recent ones from the last few days as far as I can tell; all the older ones are OK.

Edit: Just to be clear, that's all of the ones you attached to this PR; the original collection is fine.

**Comment by gusferguson** [comment]
@rleigh-dundee - sorry - needed to check box and failed. Redoing.

**Comment by gusferguson** [comment]
@rleigh-dundee @hflynn - sorry about that. I have redone them all but added _zip or _txz to appropriate versions. You may end up with a version or 2 that you don't need, but I did them all both ways in case.

bf_git_tag
bf_super-build_git_tag
bf_bioformats_repository
bf_super-build_repository
bf_c _mac_10_11_debug
bf_c _mac_10_11_release
bf_super-build_source_code_txz
bf_bioformats_source_code_txz
bf_bioformats_source_code_zip
bf_super-build_source_code_zip
bf_api_doc_zip
bf_api_doc_txz
bf_source_code_txz
bf_source_code_zip

**Comment by ghost** [comment]
@gusferguson Thanks.  @hflynn If you're happy with the latest state of the download page, I think we're done.

**Comment by hflynn** [comment]
:+1: No further issues I can see, let's call it a day for this release. Thanks Both!

**Comment by sbesson** [comment]
Thanks all

**Pull request merged**